### ISSUE TITLE
log should be written to a file

### ISSUE THREAD

**Issue by calvinfo** [calvinfo/cruise#2]

**Comment by mcollina** [comment]
I think you should use [leveldb](http://leveldb.org) instead of a standard filesystem :D, or at least provide a way to add a level backend. Level is _very_ simple to use, so... :D.

**Comment by calvinfo** [comment]
Haha, I was actually thinking about using level! :D 

The goraft/raft implementation fsync's on every 'append', so I mainly left this as a reminder so that we shouldn't forget that bit haha. Also -- syncing/reading from the existing log on startup would be pretty useful. In level I guess we could even use a .createReadStream() command or whatever gives you back a range of keys to get all the log entries. Seems like a good fit!

**Comment by jcrugzz** [comment]
I totally agree, was actually looking into this :). We can even default to just be [pure JavaScript](https://github.com/bigeasy/locket) and allow other backends to be pluggable
